bgpd/bgp_open.c

```diff
@@ -1441,35 +1440,30 @@ void bgp_open_capability(struct stream *s, struct peer *peer)
 	}
 
 	/* Hostname capability */
-	uname(&names);
-	if (names.nodename[0] != '\0') {
+	if (hostname_get()) {
 		SET_FLAG(peer->cap, PEER_CAP_HOSTNAME_ADV);
 		stream_putc(s, BGP_OPEN_OPT_CAP);
 		rcapp = stream_get_endp(s); /* Ptr to length placeholder */
 		stream_putc(s, 0);	  /* dummy len for now */
 		stream_putc(s, CAPABILITY_CODE_FQDN);
 		capp = stream_get_endp(s);
 		stream_putc(s, 0); /* dummy len for now */
-		len = strlen(names.nodename);
+		len = strlen(hostname_get());
 		if (len > BGP_MAX_HOSTNAME)
 			len = BGP_MAX_HOSTNAME;
 
 		stream_putc(s, len);
-		stream_put(s, names.nodename, len);
-#ifdef HAVE_STRUCT_UTSNAME_DOMAINNAME
-		if ((names.domainname[0] != '\0')
-		    && (strcmp(names.domainname, "(none)") != 0)) {
-			len = strlen(names.domainname);
+		stream_put(s, hostname_get(), len);
+		if ((host.domainname)
+		    && (strcmp(host.domainname, "(none)") != 0)) {
```

Comment:
Forgot to add, if the `(none)` check is in lib, the check here becomes superfluous.